Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -14,14 +14,14 @@
 
 /// Analyzes the types of variables.
 public struct AbstractInterpreter {
-    // States are kept in a stack to support conditional execution.
+    // State diffs are kept in a stacks to support conditional execution.
+    // Save final variable types changed in if block.
     private var ifStack: [VariableMap<Type>] = []
-    private var stack = [VariableMap<Type>()]
+    // Save last variable type before changing it in current block.
```

Comment:
I think here you could explain in a bit more detail how the diffs work, i.e. that they are essentially diffs from `currentState` to be able to reset/merge with a previous state.